compiler/src/lnntoamm/Types.ts

```diff
@@ -89,9 +91,27 @@ export default abstract class Type implements Equalable {
   dupIfNotLocalInterface(): Type | null {
     return null;
   }
+
+  fieldIndices(): FieldIndices {
+    let name: string;
+    try {
+      name = this.instance().name;
+    } catch (e) {
+      name = this.name;
+    }
+    if (name !== '') {
+      name = ` ${name}`;
+    }
+    throw new Error(`Type${name} does not have fields`);
+  }
```

Comment:
This is a little weird. Why does the name get a prefixed space on it? The error message could have been done by just adding a space before the dollar sign, for instance.

Reply:
because otherwise there's a duplicated space, and that was bothering me in testing :sweat_smile:

Reply:
`throw new Error(`Type${name ? ` ${name}` : ''} does not have fields`);`

Would also work (you can nest template strings, fyi) though that doesn't buy you *too* much.

You could also have something like:

```js
const formalName = name === '' ? 'Type' : `Type ${name}`;
throw new Error(`${formalName} does not have fields`);
```

But this is all extremely nitpicky over mutating the name of the type with formatting data, y'know, in case we ever want to translate the error messages into a language that doesn't put spaces between words. ;)